src/main/kotlin/ink/abb/pogo/scraper/Context.kt

```diff
@@ -35,10 +40,30 @@ data class Context(
         val pokemonInventoryFullStatus: Pair<AtomicBoolean, AtomicBoolean>,
 
         var restApiPassword: String,
+        var s2Cache: MutableMap<String, Double>,
         var restApiToken: String = "",
 
         val walking: AtomicBoolean = AtomicBoolean(false),
 
         val pauseWalking: AtomicBoolean = AtomicBoolean(false)
 
-)
+) {
+    fun getAltitude(latitude: Double, longitude: Double): Double {
+        val rand = (Math.random() * 3) + 1
+        val cellId = S2CellId.fromLatLng(S2LatLng.fromDegrees(latitude, longitude)).parent(15).id().toString()
+        if (this.s2Cache.containsKey(cellId) && this.s2Cache[cellId] != null){
+            return this.s2Cache[cellId]!! + rand
+        }
+        var elevation = 10.0
+        val url = HttpUrl.parse("https://elevation.mapzen.com/height?json={\"shape\":[{\"lat\":" + latitude.toString() + ",\"lon\":" + longitude.toString() + "}]}").newBuilder().build()
```

Comment:
Using Kotlin, you can concat it like this:
"https://elevation.mapzen.com/height?json={\"shape\":[{\"lat\":$latitude,\"lon\":$longitude}]}"